### ISSUE TITLE
[BUG] Benchmarking.run() silently overwrites results when estimators share class names - add labels parameter

### ISSUE THREAD

**Issue by Vaishnav88sk** [gc-os-ai/pyaptamer#577]
**Describe the bug**

`Benchmarking.run()` uses `estimator.__class__.__name__` as the dictionary key for results (line 132). When two estimators of the **same class** but with **different hyperparameters** are benchmarked, the second silently overwrites the first in the results dict, and only the last estimator's scores appear in the final DataFrame.

**To Reproduce**

```python
from sklearn.dummy import DummyClassifier
from sklearn.metrics import accuracy_score
from sklearn.model_selection import PredefinedSplit
import numpy as np
from pyaptamer.benchmarking._base import Benchmarking
X = np.array([[1, 0], [0, 1], [1, 1], [0, 0]] * 5)
y = np.array([1, 0, 1, 0] * 5)
test_fold = np.ones(len(y)) * -1
test_fold[-4:] = 0
cv = PredefinedSplit(test_fold)
clf1 = DummyClassifier(strategy="most_frequent")
clf2 = DummyClassifier(strategy="stratified")
bench = Benchmarking(
    estimators=[clf1, clf2],
    metrics=[accuracy_score],
    X=X, y=y, cv=cv,
)
results = bench.run()
print(results)
# Only shows ONE "DummyClassifier" row - clf1's results are lost!
```

**Expected behavior**

Both estimators should appear in the results with distinct names. The fix should:

1. Add an optional labels parameter to __init__() so users can provide custom names.
2. When labels=None, auto-detect duplicate class names and append _1, _2 suffixes.
3. Be fully backward compatible - existing code without labels still works.

**Additional context**

This is distinct from issue #543 (which only reported detection). This issue proposes a concrete fix with the labels parameter and auto-deduplication logic.

**Versions**
pyaptamer 0.1.0a1
<details>

</details>

**Comment by PixiBooth** [comment]
Hey! Cool project. I noticed you're looking for more visibility. Getting those first 100 stars is the hardest part — it's a cold start problem. If you need a boost to get initial traction, check out https://yybuu.com — we help open-source projects get the visibility they deserve. Keep up the great work!

**Comment by pranavchoudhary-tech** [comment]
Hi! I am a prospective applicant for ESoC 2026 and would like to claim this issue as part of my application.

I have implemented a fix for the benchmarking results overwrite bug by adding a `labels` parameter to the `Benchmarking.__init__` method, allowing for custom estimator naming. I've also implemented an auto-deduplication logic within the `run()` method that detects duplicate estimator class names and automatically appends `_1`, `_2` suffixes to prevent results from being silently overwritten.

I'll be opening a Pull Request with these changes shortly!

**Comment by Vaishnav88sk** [comment]
@pranavchoudhary-tech Hello Sir,

completed via #580 

Do not need to re PR this. :)

**Comment by Vaishnav88sk** [comment]
@fkiraly @NennoMP
